elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *    Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License").
+ *    You may not use this file except in compliance with the License.
+ *    A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    or in the "license" file accompanying this file. This file is distributed
+ *    on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *    express or implied. See the License for the specific language governing
+ *    permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.elasticsearch.mapping;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.ToString;
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Elasticsearch index mapping. Because there is no specific behavior for different field types,
+ * string is used to represent field types.
+ */
+@ToString
+public class IndexMapping {
+
+    /**
+     * Field mappings from field name to field type in Elasticsearch date type system.
+     */
+    private final Map<String, String> fieldMappings;
+
+    public IndexMapping(Map<String, String> fieldMappings) {
+        this.fieldMappings = fieldMappings;
+    }
+
+    public IndexMapping(MappingMetaData metaData) {
+        this.fieldMappings = flatMappings(metaData.getSourceAsMap());
+    }
+
+    /**
+     * How many fields in the index (after flatten)
+     * @return  field size
+     */
+    public int size() {
+        return fieldMappings.size();
+    }
+
+    /**
+     * Return field type by its name.
+     * @param fieldName     field name
+     * @return              field type in string
+     */
+    public String getFieldType(String fieldName) {
+        return fieldMappings.get(fieldName);
+    }
+
+    /**
+     * Get all field types and transform raw string type to expected type.
+     * @param transform     transform function to transform field type in string to another type
+     * @param <Type>        expected field type class
+     * @return              mapping from field name to field type
+     */
+    public <Type> Map<String, Type> getAllFieldTypes(Function<String, Type> transform) {
+        return fieldMappings.entrySet().
+                             stream().
+                             collect(Collectors.toMap(
+                                 Map.Entry::getKey,
+                                 e -> transform.apply(e.getValue())
```

Comment:
The null value should be filtered out.

Reply:
Thanks. Double checked there is only null value in Elasticsearch so we should be good. https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-data-types.html